Sources/TensorFlow/Core/TensorShape.swift

```diff
@@ -94,6 +94,18 @@ public extension TensorShape {
     var endIndex: Int {
         return dimensions.endIndex
     }
+    
+    static func +  (lhs: TensorShape, rhs: TensorShape) -> TensorShape {
```

Comment:
This was surprising that TensorShape doesn't behave as a standard Array or a List. That can be useful in many cases!

Reply:
The proper way to fix this is to conform `TensorShape` to `Collection`: https://github.com/tensorflow/swift-apis/issues/609.

`RangeReplaceableCollection` provides a [default implementation](https://github.com/apple/swift/blob/02d8d83658033d88c8ec99db2ee37cff99292b6b/stdlib/public/core/RangeReplaceableCollection.swift#L967) for [`+(_:_:)`](https://developer.apple.com/documentation/swift/rangereplaceablecollection/2965327).

If `TensorShape` conforms to `RangeReplaceableCollection`, manually defining `static func +` becomes 
unnecessary. I'd prefer for someone to implement the `TensorShape: RangeReplaceableCollection` conformance instead of manually defining `static func +`: would you be interested?